test/unit_tests/unittest_scanner_communication_interface.cpp

```diff
@@ -48,3 +51,36 @@ TEST(ScannerCommunicationIntefaceTests, testReadTimeout)
   auto read_buf{ boost::asio::buffer(read_array) };
   ASSERT_THROW(comm_interface->read(read_buf), ScannerReadTimeout);
 }
+
+TEST(ScannerCommunicationIntefaceTests, testScannerOpenFailedForCompleteCoverage)
+{
+  std::string except_str{ "DummyText" };
+  std::unique_ptr<ScannerOpenFailed> e(new ScannerOpenFailed(except_str));
+  EXPECT_EQ(except_str, e->what());
+}
+
+TEST(ScannerCommunicationIntefaceTests, testScannerWriteFailedForCompleteCoverage)
+{
+  std::string except_str{ "DummyText" };
+  std::unique_ptr<ScannerWriteFailed> e(new ScannerWriteFailed(except_str));
+  EXPECT_EQ(except_str, e->what());
+}
+
+TEST(ScannerCommunicationIntefaceTests, testScannerScannerReadTimeoutForCompleteCoverage)
+{
+  std::unique_ptr<ScannerReadTimeout> e(new ScannerReadTimeout());
```

Comment:
Add check that string is equal to "Timeout while waiting for message from scanner"?